docker-compose.yml

```diff
@@ -0,0 +1,36 @@
+version: '3.9'
+
+services:
+  app:
+    build: .
+    image: whitehara/fooocus
+    #ports:
+     #- 7865:7865
+    environment:
+      - CMDARGS=--share    # arguments for entry_with_updte.py
+      - DATADIR=/content/data   # directory which stores models, outputs dir
+      - config_path=/content/data/config.txt
+      - config_example_path=/content/data/config_modification_tutorial.txt
+      - path_checkpoints=/content/data/models/checkpoints/
+      - path_loras=/content/data/models/loras/
+      - path_embeddings=/content/data/models/embeddings/
+      - path_vae_approx=/content/data/models/vae_approx/
+      - path_upscale_models=/content/data/models/upscale_models/
+      - path_inpaint=/content/data/models/inpaint/
+      - path_controlnet=/content/data/models/controlnet/
+      - path_clip_vision=/content/data/models/clip_vision/
+      - path_fooocus_expansion=/content/data/models/prompt_expansion/fooocus_expansion/
+      - path_outputs=/content/app/outputs/    # Warning: If it is not under '/content/app', you can't see history log!
+      - default_model=juggernautXL_version6Rundiffusion.safetensors
+      - default_performance=Speed
+    volumes:
+      - ./data:/content/data
```

Comment:
using a virtual volume here will increase the loading speed of models in Fooocus. Please test and adjust accordingly. See https://docs.docker.com/storage/volumes/#use-a-volume-with-docker-compose
Please also consider using a mount for the outputs directory as well as the python cache dir (most likely /home/user/.cache) as keeping cache non-persistent will lead to the container having to download many models on container start after destruction again.

Reply:
#Thank you for your comment.
- Unfortunately, I can't find the word, "virtual volume" in the URL. What is the "virtual volume"?
- "models" directory is in the persistent volume. So you don't need to download them after the second boot.
- "outputs" directory is also in the persistent volume. So you can see your previous outputs after the second boot.
- Generate python precompiled cache "__pycache__" in Dockefile. ([5b30983](https://github.com/lllyasviel/Fooocus/pull/1418/commits/5b309834ba91b54adad0eb23e7b32fb83c2bd40b)). However, even without this cache, the .pyc files will be generated with a very short time.